### ISSUE TITLE
Think about display order..... 

### ISSUE THREAD

**Issue by joewheaton** [Riverscapes/RaveAddIn#14]
The RAVE toolbar does some clever things on adding to the map. It does look for the parent layer group if you move a layer group around and tries to put stuff in the right place. One thing we DO NOT do now, is consciously think about:
- Which layers make sense as true BASEMAP layers (i.e. down at bottom of TOC... generally raster basemaps)
- Which layers make sense as false BASEMAP context layers, that belong on top (e.g. generally vector things like hydrographic network, watershed boundaries, adminsistrative boundaries and other context)
- Which of "inputs" is sort of master context (e.g. hillshade), and which of inputs, intermediate and output rasters that should be transparent and on top.

## The Problem
I illustrate this in this video.
[![](http://img.youtube.com/vi/urWRqvuS4d8/0.jpg)](http://www.youtube.com/watch?v=urWRqvuS4d8 "")

## Solution?
I am wondering if we need another tag(s) in [business logic](http://rave.riverscapes.xyz/business-logic.html) to represent these things? Perhaps there is a new argument of `sortorder=""` inside each `<node >` tag? with options being:
- `sortorder = "trueBase"` - These would get put at very bottom of TOC in "Basemaps"
- `sortorder = "contextBase"`  - These would be put in "Cartographic Context" above the other layer groups that are RS project specific
- `sortorder = "masterBase"` - This would be an odd example of a layer that gets "divorced
- `sortorder = "default"` - By default and if not specified it goes on top like it does by layer groups.

I think we should also add a `transparency%` argument that by default is 0, and can be specified in any `<node   >` of business logic. This is related to #5 .

**Comment by philipbaileynar** [comment]
I completely agree. This occurred to me while I was testing. I think you have the right concepts but would prefer to come up with a better name for the layers that sit on top. Overlays?

There's also the risk of overusing terms such as "context". Context projects. Context layers in projects. Context base maps etc...

The transparency (#5 ) is needed and as I think you mentioned it isn't in the layer file so we need a business logic and base map parameter to store this. Then it should be easy (I believe the CHaMP topo toolbar used to apply transparencies so we have the code already).

**Comment by joewheaton** [comment]
I like the term "overlays". That is a nice one.

**Comment by joewheaton** [comment]
We have the same problem now over in QRAVE: https://github.com/Riverscapes/QRAVEPlugin/issues/19
